### DISCUSSION TITLE
Automatic usage of accessors

### DISCUSSION THREAD

**Discussion by pgierz** [pydata/xarray#6622] [Q&A]
Hello,

I am not sure if this is the right place for this, but I have a few questions regarding the possibility to extend xarray via dataset accessors (https://xarray.pydata.org/en/stable/internals/extending-xarray.html)

I am considering to write a small module enabling a small conversion of variable names which are (unfortunately) not in the standard GRB format. More background on that is here: https://github.com/ecmwf/cfgrib/issues/127

Would it be possible to somehow extend xarray in such a way that if one imports my module, the `open_dataset` and `open_mfdataset` functions automatically first call my little conversion tool (if needed). Something like:

```python
import xarray as xr
import pauls_conversion_lib

...magic happens via importing...

ds = xr.open_dataset("/some/path/to/special/file")
multi_file_ds = xr.open_mfdataset("/some/path/to/many/special/files")
```

I know that the holoviews community did something similar by adding the `hvplot` attribute to a dataset which works an extra on top of the standard plot. Perhaps that is the way to go?
```python
import xarray as xr
import pauls_conversion_lib

conversion_lib.activate()

ds = xr.convert_open_dataset("/some/path/to/special/file")
multi_file_ds = xr.convert_open_mfdataset("/some/path/to/many/special/files")
```

I would be happy to receive some input before I start going down the wrong rabbit hole :-)

**Comment by dcherian** [comment]
Are you just trying to rename some variables (you could provide a `preprocess` function) or is the problem that cfgrib won't actually return all the data if it doesn't know about variable names. For the latter, you could try adding a new backend: https://docs.xarray.dev/en/stable/internals/how-to-add-new-backend.html that wraps cfgriḃ.

It would be good to see a working example of what you're trying to achieve.

**Comment by aaronspring** [comment]
I think we were rather thinking about a renaming, see https://gist.github.com/aaronspring/bf5139a718bbd607dcb95a5222a9d0e8, but the backend idea sounds also intereseting. I think we need to refine our use case
